### ISSUE TITLE
"complete -C" segfaults when invoked noninteractively

### ISSUE THREAD

**Issue by jonmetz** [fish-shell/fish-shell#2361]
When running [american fuzzy lop](http://lcamtuf.coredump.cx/afl/) I found that running scripts that contain ```complete -C``` causes a segfault on Ubuntu 15.04.

For example if we have a file ```test.fish```:
```
complete -C echo
```

we get this trying to execute the file:
```
~$ fish test.fish
Segmentation fault (core dumped)
```

Opening the core file in gdb produces this:
```
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
Core was generated by `fish test.fish'.
Program terminated with signal SIGSEGV, Segmentation fault.
#0  wcslen () at ../sysdeps/x86_64/wcslen.S:24
24      ../sysdeps/x86_64/wcslen.S: No such file or directory.
```

I've reproduced this on the latest master from git and on fish from the Ubuntu repo.

**Comment by zx8** [comment]
Yep. Can confirm it's happening on OS X 10.11 with the `HEAD` version as well:

```
fish: 'fish test.fish' terminated by signal SIGSEGV (Address boundary error)
```

**Comment by ridiculousfish** [comment]
Cool tool :+1: 

`do_complete` attempts to use the command line contents if no argument is given. This seems of dubious value to me. Let's have it at least report a missing argument error in this case if there is no commandline, because we are in non-interactive mode.

**Comment by ridiculousfish** [comment]
Fixed as 025b45b91ee3cbedfa99d02c414ecfe4756cdd1f. Thanks!!
